app.py

```diff
@@ -909,14 +909,27 @@ def unescape_ptree_tok(token_list):
 					break
 
 	for i, terminal in enumerate(cgel_tree_terminals):
+
 		prepunct_token_list = prepunct_tokens[i]
 		unescape_ptree_tok(prepunct_token_list)
 		postpunct_token_list = postpunct_tokens[i]
 		unescape_ptree_tok(postpunct_token_list)
 		terminal.prepunct = prepunct_token_list
 		terminal.postpunct = postpunct_token_list
+
+		terminal.text = terminal.text.replace("_", " ")
+
 	cgel_tree.update_terminals(cgel_tree_terminals, gaps=True)
-	treestr = "(ROOT " + cgel_tree.ptb(punct=True) + ")"
+
+	cgel_tree_to_parentedtree = copy.deepcopy(cgel_tree)
+	cgel_tree_terminals_to_parentedtree = copy.deepcopy(cgel_tree_terminals)
+
+	for terminal in cgel_tree_terminals_to_parentedtree:
+		terminal.text = terminal.text.replace(" ", "_")
+
+	cgel_tree_to_parentedtree.update_terminals(cgel_tree_terminals_to_parentedtree, gaps = True)
+
+	treestr = "(ROOT " + cgel_tree_to_parentedtree.ptb(punct=True) + ")"
```

Comment:
This is pretty complicated. I guess it is a workaround to get the underscores into the PTB-style output? Looking at `cgel.ptb()` it replaces spaces with `++` (instead of `_`). I'll update that function to make it a parameter in line with `cgel.tagging()`.

Reply:
https://github.com/nert-nlp/cgel/commit/0b7e4982ad246598699190424cb61299b20aca12

Now you should be able to do `cgel_tree.ptb(punct=True, complex_lexeme_separator='_')`